### DISCUSSION TITLE
How to make reference line appear at the beginning of the bar chart instead of middle of the bar , when the scale type of x-axis is band?? Any idea?

### DISCUSSION THREAD

**Discussion by vinaypuranik** [recharts/recharts#4685] [General]
![Screenshot 2024-06-21 at 13 16 27](https://github.com/recharts/recharts/assets/6823168/7337a247-f6e6-4223-ab9a-0dba053cb933)

Want the reference line appear here as marked in teal, it appears in the middle of the bar at the moment

**Comment by ckifer** [comment]
If your XAxis is categorical the only way reference line will render is if the X value is exactly one of the categories. At this point I don't think there's a way to customize the position.

**Reply by vinaypuranik** [reply]
Nvm, Its simple, just need to pass the `position` prop

```
export declare namespace ReferenceLine {
    var displayName: string;
    var defaultProps: {
        isFront: boolean;
        ifOverflow: string;
        xAxisId: number;
        yAxisId: number;
        fill: string;
        stroke: string;
        fillOpacity: number;
        strokeWidth: number;
        position: string;
    };
}
```

**Reply by ckifer** [reply]
Oh got it, I guess I wasn't clear on the question then
